app/src/features/FhirResourceTree/resourceTreeSlice.ts

```diff
@@ -0,0 +1,75 @@
+import { IElementDefinition } from "@ahryman40k/ts-fhir-types/lib/R4";
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+import { RootState } from "app/store";
+
+export type TypeNature =
+  | "complex"
+  | "primitive"
+  | "array"
+  | "choice"
+  | undefined;
+
+type Element<T> = {
+  id: string;
+  name: string;
+  path: string;
+  nature?: TypeNature;
+  isSlice: boolean;
+  parent?: T;
+  children: T[];
+  type?: string;
+  definition: IElementDefinition;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ElementNode extends Element<ElementNode> {}
```

Comment:
```typescript 
type Element = {
  id: string;
  name: string;
  path: string;

  nature?: TypeNature;
  isSlice: boolean;

  type?: string;

  definition: IElementDefinition;
};

export interface ElementNode extends Element {
  parent?: ElementNode;
  children: ElementNode[];
}
```
could this also work?

Reply:
This doesn't, Typescript's type declaration doesn't support recursive types...